app/objects/secondclass/c_executor.py

```diff
@@ -0,0 +1,87 @@
+import marshmallow as ma
+
+from app.utility.base_object import BaseObject
+
+from app.objects.secondclass.c_parser import ParserSchema
+from app.objects.secondclass.c_variation import Variation, VariationSchema
+
+
+class ExecutorSchema(ma.Schema):
+    name = ma.fields.String(missing=None)
+    platform = ma.fields.String(missing=None)
+    command = ma.fields.String(missing=None)
+    code = ma.fields.String(missing=None)
+    language = ma.fields.String(missing=None)
+    build_target = ma.fields.String(missing=None)
+    payloads = ma.fields.List(ma.fields.String(), missing=[])
+    uploads = ma.fields.List(ma.fields.String(), missing=[])
+    timeout = ma.fields.Int(missing=60)
+    parsers = ma.fields.List(ma.fields.Nested(ParserSchema()), missing=[])
+    cleanup = ma.fields.List(ma.fields.String(), missing=[])
+    variations = ma.fields.List(ma.fields.Nested(VariationSchema()), missing=[])
+    additional_info = ma.fields.Dict(keys=ma.fields.String(), values=ma.fields.String())
+
+    @ma.post_load
+    def build_executor(self, data, **_):
+        return Executor(**data)
+
+
+class Executor(BaseObject):
+
+    schema = ExecutorSchema()
+    display_schema = ExecutorSchema()
+
+    RESERVED = dict(payload='#{payload}')
+
+    HOOKS = dict()
+
+    @property
+    def test(self):
+        """Get command with app property variables replaced"""
+        return self.decode_bytes(self.replace_app_props(self.encode_string(self.command)))
+
+    def __init__(self, name, platform, command=None, code=None, language=None, build_target=None,
+                 payloads=None, uploads=None, timeout=60, parsers=None, cleanup=None, variations=None,
+                 additional_info=None, **kwargs):
+        super().__init__()
+        self.name = name
+        self.platform = platform.lower()
+
+        self.command = command
+        self.code = code
+        self.language = language
+        self.build_target = build_target
+
+        self.payloads = payloads if payloads else []
+        self.uploads = uploads if uploads else []
+
+        self.timeout = timeout
+        self.parsers = parsers if parsers else []
+        self.cleanup = [cleanup] if cleanup else []
+        self.variations = get_variations(variations)
+
+        self.additional_info = additional_info or dict()
+        self.additional_info.update(**kwargs)
+
+    def __getattr__(self, item):
+        try:
+            return super().__getattribute__('additional_info')[item]
+        except KeyError:
+            raise AttributeError(item)
+
+    def replace_cleanup(self, command, payload):
+        return command.replace(self.RESERVED['payload'], payload)
+
+
+def get_variations(data):
+    variations = []
+    if data:
```

Comment:
Nit: mentioned elsewhere, but using guard clauses is nice and reduces nesting (https://wiki.c2.com/?GuardClause)

So right at the top of the function, just do

```python
if not data:
    return []
```

Reply:
(Just noticed you copied it, but we could still clean up as we go!)